### ISSUE TITLE
error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

### ISSUE THREAD

**Issue by rmib200** [juicycleff/flutter-unity-view-widget#581]
**Describe the bug**
After importing the unitypackage a "Multiple precompiled assemblies with the same name Newtonsoft.Json.dll included or the current platform. Only one assembly with the same name is allowed per platform." error appeared. I deleted the JsonDotNet folder and event tried to impot the unitypackage without the folder but the same errors appears:

Assets\FlutterUnityIntegration\NativeAPI.cs(29,13): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(29,39): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(32,16): error CS0246: The type or namespace name 'Exception' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(34,13): error CS0103: The name 'Debug' does not exist in the current context

Assets\FlutterUnityIntegration\NativeAPI.cs(48,13): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(48,39): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(51,16): error CS0246: The type or namespace name 'Exception' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(53,13): error CS0103: The name 'Debug' does not exist in the current context

Assets\FlutterUnityIntegration\NativeAPI.cs(67,26): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(68,49): error CS0246: The type or namespace name 'AndroidJavaObject' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(71,16): error CS0246: The type or namespace name 'Exception' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(73,13): error CS0103: The name 'Debug' does not exist in the current context

Assets\FlutterUnityIntegration\NativeAPI.cs(85,13): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(85,39): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(86,13): error CS0246: The type or namespace name 'AndroidJavaObject' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(86,63): error CS0246: The type or namespace name 'AndroidJavaObject' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(89,16): error CS0246: The type or namespace name 'Exception' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(91,13): error CS0103: The name 'Debug' does not exist in the current context

Assets\FlutterUnityIntegration\NativeAPI.cs(103,13): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(103,39): error CS0246: The type or namespace name 'AndroidJavaClass' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(104,13): error CS0246: The type or namespace name 'AndroidJavaObject' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(104,63): error CS0246: The type or namespace name 'AndroidJavaObject' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(107,16): error CS0246: The type or namespace name 'Exception' could not be found (are you missing a using directive or an assembly reference?)

Assets\FlutterUnityIntegration\NativeAPI.cs(109,13): error CS0103: The name 'Debug' does not exist in the current context

**To Reproduce**
1. Import Package as directed in readme
2. Delete JsonDotNet folder
3. New errors appeared

**Expected behavior**
A "Flutter" option should appear in the navbar.

**Screenshots**

**Unity (please complete the following information):**
 - OS: Windows 10
 - Version 2020.3.28f1

**Smartphone (please complete the following information):**
 
 - OS: Android
 

**Additional context**

**Comment by WouterVandenputte** [comment]
Same issue here. (Also het to delete the jsonDotNet since it conflicted with another dll in the `Library` folder). Can't seem to remember it was there in previous versions so it might be a new bug to be found in one of the more recent commits

**Comment by rmib200** [comment]
Thanks! I helped solve many of the problems. Although 2 persist:

Assets\FlutterUnityIntegration\Editor\Build.cs(276,37): error CS0117: 'EditorUserBuildSettings' does not contain a definition for 'iOSXcodeBuildConfig'

Assets\FlutterUnityIntegration\Editor\Build.cs(276,59): error CS0103: The name 'XcodeBuildConfig' does not exist in the current context

Did you also encountered them after adding the imports?. I'm actually building for android and my project is set for that platform.

**Comment by ernonemacher** [comment]
I believe these errors are happening in the newest versions of Unity. In order to fix it and run normally I did what @WouterVandenputte said and changed the Build.cs file from

`EditorUserBuildSettings.iOSXcodeBuildConfig = XcodeBuildConfig.Release;`

to

`EditorUserBuildSettings.iOSBuildConfigType = iOSBuildType.Release;`

**Comment by rmib200** [comment]
Thanks you are a savior!

> I believe these errors are happening in the newest versions of Unity. In order to fix it and run normally I did what @WouterVandenputte said and changed the Build.cs file from
> 
> `EditorUserBuildSettings.iOSXcodeBuildConfig = XcodeBuildConfig.Release;`
> 
> to
> 
> `EditorUserBuildSettings.iOSBuildConfigType = iOSBuildType.Release;`

**Comment by rmib200** [comment]
I was trying to export my project going to "Flutter/Export Android", but when it is about to finish this error appears: 


FileNotFoundException: Could not find file "D:\Unity\android\build.gradle"
System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.IO.FileOptions options, System.String msgPath, System.Boolean bFromProxy, System.Boolean useLongPath, System.Boolean checkHost) (at <695d1cc93cca45069c528c15c9fdd749>:0)
(wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int,System.IO.FileOptions,string,bool,bool,bool)
System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding, System.Boolean detectEncodingFromByteOrderMarks, System.Int32 bufferSize, System.Boolean checkHost) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding, System.Boolean detectEncodingFromByteOrderMarks, System.Int32 bufferSize) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.IO.StreamReader..ctor (System.String path, System.Boolean detectEncodingFromByteOrderMarks) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.IO.StreamReader..ctor (System.String path) (at <695d1cc93cca45069c528c15c9fdd749>:0)
(wrapper remoting-invoke-with-check) System.IO.StreamReader..ctor(string)
System.IO.File.ReadAllText (System.String path) (at <695d1cc93cca45069c528c15c9fdd749>:0)
FlutterUnityIntegration.Editor.Build.SetupAndroidProject () (at Assets/FlutterUnityIntegration/Editor/Build.cs:356)
FlutterUnityIntegration.Editor.Build.DoBuildAndroid (System.String buildPath, System.Boolean isPlugin) (at Assets/FlutterUnityIntegration/Editor/Build.cs:171)
FlutterUnityIntegration.Editor.Build.DoBuildAndroidLibrary () (at Assets/FlutterUnityIntegration/Editor/Build.cs:33)


I tried enabling "Export project" in Build Settings and enabling custom gradle in Publisher Settings with no success. 
Does anyone has any idea on how to proceed?

**Comment by jamesncl** [comment]
@rmib200 I think this is a separate issue than the original one you posted so maybe worth opening a separate issue to keep things more organised. To answer your question - looks like the script can't find your gradle files so are you sure you have the correct folder structure?  I think it needs to be:

- project root   <-- the root of your flutter project
-- android  <-- android project
-- ios    <-- ios project
-- lib    <-- flutter / dart files
-- unity
---- your unity project root

PS please keep this issue open so that @juicycleff can fix the original issue, which is that the unity package is missing the `System` and `UnityEngine` imports from `NativeApi.cs`

**Comment by timbotimbo** [comment]
The orignal issue with the NativeApi compile errors was fixed in `fuw-2022.1.1-v2.unitypackage`.

**Comment by rmib200** [comment]
I'm closing the issue

> The orignal issue with the NativeApi compile errors was fixed in `fuw-2022.1.1-v2.unitypackage`.

**Comment by lightlessdays** [comment]
Double click on the error. A C# file will open in Visual Studio. Delete all the namespaces from the file and add the following namespaces-
```
using System;
using UnityEngine;
using UnityEngine.SceneManagement;
```
